mne/preprocessing/maxwell.py

```diff
@@ -1973,8 +2001,23 @@ def find_bad_channels_maxwell(
         if pick in params['grad_picks'] else
         flat_limits['mag']
         for pick in good_meg_picks])
+
     flat_step = max(20, int(30 * raw.info['sfreq'] / 1000.))
     all_flats = set()
+
+    # Prepare variables to return if `return_scores=True`.
+    bin_edges = np.r_[starts, stops[-1]]  # Ensure we include the endpoint!
```

Comment:
This is only correct if the data has no BAD annotations in it. So it would be better to keep the `starts` and `stops` rather than go with `bin_edges`.